assets/js/src/form_editor/store/blocks_to_form_body.jsx

```diff
@@ -2,15 +2,29 @@
 const mapCustomField = (block, customFields, mappedCommonProperties) => {
   const customField = customFields.find((cf) => cf.id === block.attributes.customFieldId);
   if (!customField) return null;
-  return {
+  const typesMap = {
+    'mailpoet-form/custom-text': 'text',
+    'mailpoet-form/custom-textarea': 'textarea',
+    'mailpoet-form/custom-radio': 'radio',
+  };
+  const mapped = {
     ...mappedCommonProperties,
     id: block.attributes.customFieldId.toString(),
     name: customField.name,
-    params: {
-      ...mappedCommonProperties.params,
-      validate: block.attributes.validate,
-    },
+    type: typesMap[block.name],
   };
+  if (Object.prototype.hasOwnProperty.call(block.attributes, 'validate')) {
```

Comment:
Maybe we could use `https://lodash.com/docs/4.17.15#has` instead of `Object.prototype.hasOwnProperty` here